src/app/pages/sign-stacks-message-request/use-sign-stacks-message-request.ts

```diff
@@ -1,41 +1,64 @@
-import { useCallback, useState } from 'react';
-
-import { ClarityValue, TupleCV, createStacksPrivateKey } from '@stacks/transactions';
+import { useState } from 'react';
 
 import { finalizeMessageSignature } from '@shared/actions/finalize-message-signature';
-import { signMessage, signStructuredDataMessage } from '@shared/crypto/sign-message';
 import { logger } from '@shared/logger';
-import { UnsignedMessage, whenSignableMessageOfType } from '@shared/signature/signature-types';
-import { isString } from '@shared/utils';
+import {
+  UnsignedMessage,
+  isStructuredMessageType,
+  isUtf8MessageType,
+  whenSignableMessageOfType,
+} from '@shared/signature/signature-types';
 
 import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
+import {
+  getSignaturePayloadFromToken,
+  getStructuredDataPayloadFromToken,
+} from '@app/common/signature/requests';
 import { useWalletType } from '@app/common/use-wallet-type';
 import { createDelay } from '@app/common/utils';
 import { useLedgerNavigate } from '@app/features/ledger/hooks/use-ledger-navigate';
-import { useCurrentStacksAccount } from '@app/store/accounts/blockchain/stacks/stacks-account.hooks';
+import {
+  StructuredPayload,
+  Utf8Payload,
+} from '@app/features/stacks-message-signer/stacks-message-signing';
+import { useMessageSignerStacksSoftwareWallet } from '@app/features/stacks-message-signer/stacks-message-signing.utils';
 import { useSignatureRequestSearchParams } from '@app/store/signatures/requests.hooks';
 
+// TODO: We sure we want users to wait for 1 second for nothing?
```

Comment:
Without this, you click accept and it insta-closes, which happens so fast it almost seems like something's broken. 

The delay allows time to see the loading spinner and a toast.